app/perfunwind.cpp

```diff
@@ -682,6 +687,36 @@ void PerfUnwind::analyze(const PerfRecordSample &sample)
     sendBuffer(buffer);
 }
 
+void PerfUnwind::updateIpCache()
+{
+    const auto& stack = m_currentUnwind.ipStack;
+    for (int i = 0; i < stack.length(); i++) {
+        m_ipCache[m_currentUnwind.sample->pid()][stack[i].address] = {m_currentUnwind.frames, stack[i].frameOffset};
```

Comment:
cache the cache :) otherwise this does a hash lookup in every iteration. also use range-based for:

```
auto &ipCache = m_ipCache[m_currentUnwind.sample->pid()];
for (const auto &stack : std::as_const(m_currentUnwind.ipStack)) {
    ipCache[stack.address] = {m_currentUnwind.frames, stack.frameOffset};
}
```